jujugui/static/gui/src/app/components/profile/profile.js

```diff
@@ -0,0 +1,80 @@
+/* Copyright (C) 2017 Canonical Ltd. */
+
+'use strict';
+
+class Profile extends React.Component {
+
+  constructor(props) {
+    super(props);
+    const activeSection = this.props.activeSection || 'models';
+    this.state = {activeSection};
+  }
+
+  componentWillReceiveProps(nextProps) {
+    this.setState({activeSection: nextProps.activeSection});
+  }
+
+  shouldComponentUpdate(nextProps, nextState) {
+    if (nextProps.activeSection !== this.state.activeSection) {
+      return true;
+    }
+    return false;
+  }
+
+  _generateContent() {
```

Comment:
I appears that we can reuse the map I suggested above again and again, and avoid all the repetition we have currently, even when calculating the default section if not provided above (shuld be the first entry in the map, rather than 'models'). So maybe define the map here and then propagate it to Navigation? I'd like to avoid having to write the same logic multiple time when adding a section. I think I also like the navigation to be as agnostic as possible about the number and labels of sections.